blog/06-02-2023-catering-to-the-client-side.md

```diff
@@ -61,6 +61,39 @@ sequenceDiagram
 
 Put another way, with client-side feature flagging we can separate flag [**evaluation**](/docs/specification/glossary#evaluating-flag-values) - passing an evaluation context through a set of rules in order to determine a flagging decision - from flag [**resolution**](/docs/specification/glossary#resolving-flag-values) - getting the flagging decision for a specific feature flag.
 
+### Keeping rulesets in sync for local-evaluated systems
+
+Flagging frameworks that use a local evaluation model don't have to contend with network calls for every evaluation, but they still need to keep their local rulesets up to date and allow the client-side app to respond to changes in those ruleset. Again, this means using caches to keep the most recent ruleset available. It also means that the flagging framework may have an eventing or callback mechanism to inform application code that the ruleset has changed and that current flag values may be stale.
+
+```mermaid
+sequenceDiagram
+  actor app
+  participant sdk as feature flag SDK
+  participant cache as local ruleset
+  participant provider as flag service
+  app->>+sdk: boot
+  sdk->>+provider: get ruleset from last known state
+  provider-->>-sdk: ruleset
+  sdk->>+cache: Store ruleset
+  cache->>-sdk: .
+  sdk->>-app: .
+  app->>+sdk: operation that needs a flagging decision
+  sdk->>+cache: flag evaluation
+  cache->>-sdk: locally evaluated flag value
+  sdk->>-app: flag value 
+  note left of provider: some time later, flagging rules have been updated
```

Comment:
You should consider including this in the eager evaluation diagram as well. Both techniques can support cache busting.